onnx/defs/quantization/defs.cc

```diff
@@ -145,38 +145,57 @@ y = saturate (round (x / y_scale) + y_zero_point)
 
 ONNX_OPERATOR_SET_SCHEMA(
     DynamicQuantizeLinear,
-    11,
+    13,
     OpSchema()
-        .SetDoc(DynamicQuantizeLinear_ver11_doc)
+        .SetDoc(DynamicQuantizeLinear_ver13_doc)
         .Input(0, "x", "Input tensor", "T1")
         .Output(0, "y", "Quantized output tensor", "T2")
         .Output(1, "y_scale", "Output scale. It's a scalar, which means a per-tensor/layer quantization.", "tensor(float)")
         .Output(2, "y_zero_point", "Output zero point. It's a scalar, which means a per-tensor/layer quantization.", "T2")
+        .Attr(
+            "axis",
+            "(Optional) The axis of the quantization dimension of the input tensor. Negative value means counting dimensions from the back. Accepted range is [-r, r-1] where r = rank(input)",
+            AttributeProto::INT,
+            static_cast<int64_t>(1))
         .TypeConstraint(
           "T1",
           {"tensor(float)"},
           "Constrain 'x' to float tensor.")
-      .TypeConstraint(
+        .TypeConstraint(
           "T2",
           {"tensor(uint8)"},
           "Constrain 'y_zero_point' and 'y' to 8-bit unsigned integer tensor.")
-      .FunctionBody(FunctionBodyHelper::BuildNodes(
-          {// nodes: {outputs, op, inputs, attributes}
-           FunctionBodyHelper::Const<float>("Q_Min", 0.f),
-           FunctionBodyHelper::Const<float>("Q_Max", 255.f),
-           {{"X_Min"}, "ReduceMin", {"x"}, {MakeAttribute("keepdims", int64_t(0))}},
-           {{"X_Min_Adjusted"}, "Min", {"X_Min", "Q_Min"}},
-           {{"X_Max"}, "ReduceMax", {"x"}, {MakeAttribute("keepdims", int64_t(0))}},
-           {{"X_Max_Adjusted"}, "Max", {"X_Max", "Q_Min"}},
-           {{"X_Range"}, "Sub", {"X_Max_Adjusted", "X_Min_Adjusted"}},
-           {{"Scale"}, "Div", {"X_Range", "Q_Max"}},
-           {{"Min_Scaled"}, "Div", {"X_Min_Adjusted", "Scale"}},
-           {{"Initial_ZeroPoint_FP"}, "Sub", {"Q_Min", "Min_Scaled"}},
-           {{"Clipped_ZeroPoint_FP"}, "Clip", {"Initial_ZeroPoint_FP", "Q_Min", "Q_Max"}},
-           {{"Rounded_ZeroPoint_FP"}, "Round", {"Clipped_ZeroPoint_FP"}},
-           {{"Zeropoint"}, "Cast", {"Rounded_ZeroPoint_FP"}, {MakeAttribute("to", int64_t(2))}},
-           {{"y_scale"}, "Identity", {"Scale"}},
-           {{"y_zero_point"}, "Identity", {"Zeropoint"}},
-           {{"y"}, "QuantizeLinear", {"x", "Scale", "Zeropoint"}}})));
+        .SetContextDependentFunctionBodyBuilder(
+            [](const FunctionBodyBuildContext& ctx,
+                const OpSchema& schema,
+                FunctionProto& functionProto) -> bool {
+                const auto axis = ctx.getAttribute("axis") != nullptr ? ctx.getAttribute("axis")->i() : OPTIONAL_VALUE;
```

Comment:
nit: DEFAULT_VALUE

Reply:
OPTIONAL_VALUE is a global variable (`false`) which is the default value of "axes" attribute in Reduction functions. Actually I don't know what the proper default value here is if the "axis" attribute does not specify in `DynamicQuantizeLinear`. I believe this is why `test_dynamicquantizelinear_min_adjusted_expanded_cpu` is failing.

Reply:
The quantize op spec says that the default value is 1, doesn't it?

Reply:
We should be using the default (1) for DynamicQuantizeLinear (not from Reduction ops).

Reply:
If that is the case, it seems that it will break 3 existing test cases... I want to make sure one thing:
1. (Original) ReduceMax and ReduceMin (keepdims=0, axes is not specified; which means reduce all dimensions)
2. (Updated) ReduceMax and ReduceMin (axes=1, keepdims=0, complement_axes=1)

Will these default cases, 1 and 2, produce the same shape?